jupyterhub/templates/hub/deployment.yaml

```diff
@@ -75,8 +75,8 @@ spec:
           persistentVolumeClaim:
             claimName: {{ include "jupyterhub.hub-pvc.fullname" . }}
         {{- end }}
-      {{- if .Values.rbac.enabled }}
-      serviceAccountName: {{ include "jupyterhub.hub.fullname" . }}
+      {{- with include "jupyterhub.hub-serviceaccount.fullname" . }}
+      serviceAccountName: {{ . }}
       {{- end }}
```

Comment:
Right now with this logic, we will provide a serviceAccountName if `.name` is specified or `.create` is set, but not if `.name` isn't set and `.create` isn't set.

We are all in agreement that we want to be allowed to set serviceAccountName if `.name` is specified, no matter if `.create` is set which we now do. Is https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/2736#discussion_r888137126 also a request for something additional besides this?